Add some recent expression functions (#5452)
qgis/QGIS-Documentation: docs/user_manual/working_with_vector/expression.rst

   "force_rhr", "Forces a geometry to respect the Right-Hand-Rule (see also
   :ref:`qgisforcerhr`)"
   "geom_from_gml", "Returns a geometry created from a GML representation of geometry"
   "geom_from_wkb |312|", "Returns a geometry created from a Well-Known Binary (WKB) representation"
   "geom_from_wkt", "Returns a geometry created from a well-known text (WKT) representation"
   "geom_to_wkb |312|", "Returns the Well-Known Binary (WKB) representation of

>    a geometry as a binary blob"

Author: this is taken directly from the software... if you want to report...

>    a geometry as a binary blob"

Reviewer: We should strive to help our users by formulating things as good as we can in the documentation, and not always just copy text from the source code / software.

>    a geometry as a binary blob"

Author: Don't we? :angry: This is a bit offending.

Now, It's even better if the (well) formulated text was reported back to the code so that it benefits to ALL users.

>    a geometry as a binary blob"

Reviewer: > Now, It's even better if the (well) formulated text was reported back to the code so that it benefits to ALL users.

I agree that reporting back is beneficial. But we should not delay making the documentation as good as we can until our suggestions are be applied in code.

>    a geometry as a binary blob"

Reviewer: > Don't we? angry This is a bit offending.

It is easy to just copy from the code - we do that all the time. The text from the code is a good starting point, but we should work with the text to make it as easy to understand as possible. Not all the coders are native English speakers, and it varies how much effort they put into "documenting" their code.

>    a geometry as a binary blob"

Author: > I agree that reporting back is beneficial. But we should not delay making the documentation as good as we can until our suggestions are be applied in code.

The suggestion was committed before we even begin this discussion. So I can't see what that would delay anything. My suggestion was to report back/open a PR so that the GUI where most of the users will meet this function provides also a clearer information.

>    a geometry as a binary blob"

Reviewer: Then everything is fine.  I thought that you did not want to accept my suggestion because it was not identical to what was written in the "code". Sorry for misunderstanding - again.

>    a geometry as a binary blob"

Reviewer: https://github.com/qgis/QGIS/pull/36327

   :widths: 20, 75
   :class: longtable

>    "ascii |314|", "Returns the unicode code associated with the first character of a string"

Reviewer: "ascii" sounds like a strange name for something that produces Unicode

Author: Taken from QGIS, it's new so if you feel to suggest a new name, it can be changed/applied before 3.4 release

Reviewer: I checked #35839, and it refers to "PostgreSQL definition of the function". That definition is "ASCII code of the first character of the argument. For UTF8 returns the Unicode code point of the character. For other multibyte encodings, the argument must be an ASCII character". We must use that definition,

>    "ascii |314|", "Returns the unicode code associated with the first character of a string"

Reviewer: ```suggestion
   "ascii |314|", "Returns the ASCII code of the first character of the argument. For UTF8 returns the Unicode code point of the character. For other multibyte encodings, the argument must be an ASCII character."
```
This should be the correct definition, taken from postgresql 10.0 documentation - https://www.postgresql.org/docs/10/functions-string.html

Author: Does the implementation in QGIS cover all the cases?

Reviewer: > Does the implementation in QGIS cover all the cases?

According to https://github.com/qgis/QGIS/pull/35839, yes.

Author: Let's ask devs directly and skip discussion.
@nyalldawson @JanCaha sorry to ping but does the suggestion above reflects what is implemented in QGIS? Thanks.

Reviewer: I have added a comment in https://github.com/qgis/QGIS/pull/35839#issuecomment-626333393

Author: With no feedback here and there, I'll just move on with what I'm sure of and if any more information, someone else would handle that.
Merging

   "item_uuid", "Returns the layout item unique ID"
   "layer", "Returns the current layer"
   "layer_id", "Returns the ID of current layer"

>    "layer_ids |314|", "Returns the list of all map layer IDs from the current project"

Reviewer: ```suggestion
   "layer_ids |314|", "Returns the IDs of all the map layers in the current project as a list"
```

   "layer_name", "Returns the name of current layer"

>    "layers |314|", "Returns the list of all map layers from the current project"

Reviewer: ```suggestion
   "layers |314|", "Returns all the map layers in the current project as a list"
```
